### PR TITLE
[add] model config, [delete] json config, [update] modules

### LINKED ISSUE #43: Too many json files
지금 Json 파일이 너무 많고, 테스트용 json 파일도 너무 많은거 같은데 
baseline처럼 어느정도 고정된 모델이나, 진행하고 있는 중요한 파일 위주로 한번 정리를 하면 어떨지 싶습니다.

### PR DESCRIPTION
**Pull request by lsg1213**
resolved #43

### CODE REVIEW

File: feature_extractor.py [hunk 1]
```diff
@@ -274,20 +274,21 @@ def polar_to_cartesian(coordinates):
 if __name__ == '__main__':
     # How to use
     # Extracting Features and Labels
+    mode = 'foa'
```
Comment (daniel03c1) [inline feature_extractor.py]: mode를 입력받는건 되게 좋네요 혹시 이거를 argparse로 해서 받도록 하는건 어떤가요?

File: train.py [hunk 2]
```diff
@@ -91,19 +95,24 @@ def iterloop(model, dataset, sed_loss, doa_loss, metric_class, config, epoch, wr
...
             lambda x, y: (mask(x, axis=-3, max_mask_size=config.time_mask_size, n_mask=6), y),
             lambda x, y: (mask(x, axis=-2, max_mask_size=config.freq_mask_size), y),
         ]
     else:
         sample_transforms = []
-    batch_transforms = [
-        split_total_labels_to_sed_doa
-    ]
+    batch_transforms = [split_total_labels_to_sed_doa]
+    if 'aug' in config.name and mode == 'train':
```
Comment (daniel03c1) [inline train.py]: config.name에 aug라는 키워드가 있는지 여부에 따라서 학습 패턴이 달라지는건
나중에라도 의도하지 않았지만, 이름에 aug가 안들어가서 세팅값과 다르게 돌아가도 나중에 파악이 되니까
옵션과 무관하게 config.name 등에도 영향을 받는 것은 조금 피하면 어떨까 싶습니다.

가령 이름을 test로 가볍게 테스트해보고 싶은데, config.name에 aug가 안들어가서 augmentation이 안들어가거나
그런 문제가 생길 것 같습니다. 이름보다는 옵션으로 주는게 좋을 것 같습니다.

File: train.py [hunk 3]
```diff
@@ -186,19 +196,22 @@ def main(config):
...
+            early_stop_patience = 0
+            lr_decay_patience = 0
             tf.keras.models.save_model(
                 model, 
                 os.path.join(model_path, f'bestscore_{best_score}.hdf5'), 
                 include_optimizer=False)
         else:
-            # TODO: reduce lr on plateau
-            if patience == 80 % config.loop_time and config.decay != 1 and config.model != 'seldnet':
+            if lr_decay_patience == config.lr_patience and config.decay != 1:
```
Comment (daniel03c1) [inline train.py:206]: 사실 이쪽 파트로 인해서 여러가지 learning rate scheduler을 적용해보고 싶은데
조금 제약이 있거나 복잡해지는 것 같더라고요, learning rate scheduler는 따로 빼는 방식으로 구현해보는건 어떤가요?

File: train.py [hunk 4]
```diff
@@ -186,19 +196,22 @@ def main(config):
...
                 optimizer.learning_rate = optimizer.learning_rate * config.decay
-            if patience == config.patience:
+                print(f'lr: {optimizer.learning_rate.numpy()}')
+                lr_decay_patience = 0
+            if early_stop_patience == config.patience // config.loop_time:
```
Comment (daniel03c1) [inline train.py]: 여기 뿐만 아니라 params.py에도 해당되는것 같은데
config.loop_time하고 무관하게 하는건 어떨까요? 즉 그냥 config.patience로 하는거죠

config.loop_time은 단지 한 epoch의 사이즈를 조절하는 거고, 보통 early stop patience는 몇번의 epoch 동안
참을 것인가를 결정하는데, step 수로 하지 않는 이상 이것은 epoch동안 몇 step이 들어있냐와 무관하게 
돌아가는 파트이기 때문에 조금 더 직관적이고, 
patience를 계산할 때 loop_time까지 고려해야하는데, 직관적이지는 않은 것 같아서 누락되거나 그래서
조금 다른 결과가 나온다거나, 다시 돌려야하거나 그런 상황이 생기지 않을까 싶습니다.

File: feature_extractor.py (hunk 1 above)
Comment (lsg1213) [reply feature_extractor.py]: 넵 그렇게 진행해서 수정할게요

Comment (lsg1213) [reply feature_extractor.py]: argparse를 사용할거면 아예 gpus도 받아버릴까요?

File: train.py (hunk 2 above)
Comment (lsg1213) [reply train.py]: 아 이건 제 개인 편의로 했던건데 말씀해주신대로 수정할게요

File: train.py (hunk 3 above)
Comment (lsg1213) [reply train.py:206]: 아 알겠습니다 이건 1등팀 스케줄러 그대로 들고온거였거든요

File: train.py (hunk 4 above)
Comment (lsg1213) [reply train.py]: 넵 알겠습니다 저부분도 수정할게요 1등팀 세팅에 맞게 하다보니 저런 설계가 있었네요

File: train.py (hunk 3 above)
Comment (lsg1213) [reply train.py:206]: issue에 올렸습니다.

**Review by daniel03c1 — APPROVED** [review]

**Review by ParkJinHyeock — APPROVED** [review]
머지 하겠습니다
